### PR TITLE
[Frontend][Relay] Keras softmax and prelu fix under NHWC

### PR DESCRIPTION
**Pull request by domin1985**
Hi,
Tvm error occurred when I imported a pre-trained keras model with NHWC layout:

TVMError:
Error(s) have occurred. The program has been annotated with them:
In main:
%0 = negative(%v_param_3);
%1 = nn.conv2d(%input_1, %v_param_1, padding=[4, 4], channels=64, kernel_size=[9, 9], data_layout="NHWC", kernel_layout="HWIO");
%2 = nn.bias_add(%1, %v_param_2, axis=-1);
%3 = negative(%2);
%4 = nn.relu(%3);
%5 = multiply(%0, %4) Incompatible broadcast type TensorType([64, 1, 1], float32) and TensorType([1, (int64)96, (int64)96, 64], float32); ;
%6 = nn.relu(%2);

softmax and prelu convertion should take layout into consideration.
@siju-samuel @yongwww @jwfromm

### CODE REVIEW

File: tests/python/frontend/keras/test_forward.py [hunk 1]
```diff
@@ -182,6 +182,7 @@ def test_forward_activations(self, keras):
             x = act_func(data)
             keras_model = keras.models.Model(data, x)
             verify_keras_frontend(keras_model)
+            verify_keras_frontend(keras_model, False, 'NHWC')
```
Comment (leandron) [inline tests/python/frontend/keras/test_forward.py:185]: minor suggestion: `verify_keras_frontend` will automatically set `need_transpose` to `false` when the layout is not `NCHW`. So in this case if you want to make it explicit, I suggest naming the parameters.

```
verify_keras_frontend(keras_model, need_transpose=False, layout='NHWC')
```

File: python/tvm/relay/frontend/keras.py [hunk 2]
```diff
@@ -123,10 +124,11 @@ def _convert_advanced_activation(inexpr, keras_layer, etab):
         if isinstance(axis, list):
             raise tvm.error.OpAttributeUnImplemented(
                 'Softmax with axes {} is not supported.'.format(axis))
-        if axis == -1:
-            axis = 1
-        else:
-            axis = axis + 1 if axis < dims - 1 else 1
+        if etab.data_layout == 'NCHW':
```
Comment (leandron) [inline python/tvm/relay/frontend/keras.py:127]: Maybe silly question but need to ask: are there no expected adjusts to be made on `axis` here when layout is not `NCHW`?

Comment (domin1985) [reply python/tvm/relay/frontend/keras.py:127]: TVM Keras frontend supports tensorflow backend only and the data_format attribute is channel_last (aka NHWC or NDHWC). So when the target layout is set NCHW (as default), the axis value should be transformed to be consist with the target layout.

File: tests/python/frontend/keras/test_forward.py (hunk 1 above)
Comment (domin1985) [reply tests/python/frontend/keras/test_forward.py:185]: OK, that's better.

File: python/tvm/relay/frontend/keras.py (hunk 2 above)
Comment (leandron) [reply python/tvm/relay/frontend/keras.py:127]: Understood - thanks

**Review by yongwww — APPROVED** [review]
LGTM

**Review by kazum — APPROVED** [review]

**Comment by kazum** [comment]
@domin1985 The change looks good. Could you rebase against the master to trigger the CI?

**Comment by kazum** [comment]
Thanks @domin1985 @leandron @jwfromm @yongwww !
